libclc/amdgcn-amdhsa/libspirv/atomic/atomic_helpers.h

```diff
@@ -71,3 +73,42 @@
   AMDGPU_ATOMIC_IMPL(FUNC_NAME, TYPE, TYPE_MANGLED, global, U3AS1, 1, BUILTIN) \
   AMDGPU_ATOMIC_IMPL(FUNC_NAME, TYPE, TYPE_MANGLED, local, U3AS3, 1, BUILTIN)  \
   AMDGPU_ATOMIC_IMPL(FUNC_NAME, TYPE, TYPE_MANGLED, , , 0, BUILTIN)
+
+#define AMDGPU_SAFE_ATOMIC_IMPL(FUNC_NAME, TYPE, TYPE_MANGLED, AS, AS_MANGLED,                                                                                         \
+                                SUB1, SUB2, BUILTIN, OP)                                                                                                               \
+  _CLC_DEF TYPE                                                                                                                                                        \
+      _Z29__spirv_AtomicCompareExchangeP##AS_MANGLED##TYPE_MANGLED##N5__spv5Scope4FlagENS##SUB1##_19MemorySemanticsMask4FlagES##SUB2##_##TYPE_MANGLED##TYPE_MANGLED(   \
+          volatile AS TYPE *, enum Scope, enum MemorySemanticsMask,                                                                                                    \
+          enum MemorySemanticsMask, TYPE desired, TYPE expected);                                                                                                      \
+  _CLC_DEF TYPE                                                                                                                                                        \
+      _Z18__spirv_AtomicLoadP##AS_MANGLED##K##TYPE_MANGLED##N5__spv5Scope4FlagENS1_19MemorySemanticsMask4FlagE(                                                        \
+          const volatile AS TYPE *, enum Scope, enum MemorySemanticsMask);                                                                                             \
+  _CLC_DEF TYPE                                                                                                                                                        \
+      FUNC_NAME##P##AS_MANGLED##TYPE_MANGLED##N5__spv5Scope4FlagENS##SUB1##_19MemorySemanticsMask4FlagE##TYPE_MANGLED(                                                 \
+          volatile AS TYPE *p, enum Scope scope,                                                                                                                       \
+          enum MemorySemanticsMask semantics, TYPE val) {                                                                                                              \
+    int atomic_scope = 0, memory_order = 0;                                                                                                                            \
+    GET_ATOMIC_SCOPE_AND_ORDER(scope, atomic_scope, semantics, memory_order)                                                                                           \
+    if (__oclc_amdgpu_reflect("AMDGPU_OCLC_UNSAFE_XOR"))                                                                                                               \
```

Comment:
See here for use of `__oclc_amdgpu_reflect`. This macro func can become more generic if we have other uses for the reflect